pynicotine/downloads.py

```diff
@@ -1042,20 +1048,39 @@ def retry_downloads(self, downloads):
             self.retry_download(download, bypass_filter)
 
     def retry_download_limits(self):
+        """Re-queue limited downloads every 5 minutes."""
+
+        if not self.limited_user_times:
+            return
 
+        current_time = time.monotonic()
+        ready_users = set()
         limited_statuses = {"Too many files", "Too many megabytes"}
 
+        for username, time_limited in self.limited_user_times:
```

Comment:
If we are checking this every second, then this loop is not required.

It would seem to be satisfactory enough to check if just the oldest item is 12 minutes old or not.

Reply:
Looks like I forgot the timer values are in seconds. The timer is supposed to run every 60 seconds.